src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -240,22 +265,36 @@ public int hashCode() {
     public static ModelMetadata fromString(String modelMetadataString) {
         String[] modelMetadataArray = modelMetadataString.split(DELIMITER, -1);
 
-        if (modelMetadataArray.length != 7) {
+        // Training node assignment was added as a field in Version 2.12.0
+        // Because models can be created on older versions and the cluster can be upgraded after,
+        // we need to accept model metadata arrays both with and without the training node assignment.
+        if (modelMetadataArray.length == 7) {
+            log.info("Model metadata array does not contain training node assignment. Assuming empty string.");
```

Comment:
Let's switch to log.debug. We don't want to pollute log file as much as possible.